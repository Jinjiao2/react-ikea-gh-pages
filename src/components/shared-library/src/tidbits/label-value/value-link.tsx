import React, { ReactNode } from 'react'

import { Text, Link } from '@tidbits/react-tidbits'

import { DEFAULT_FALLBACK_TEXT } from '../helpers'
import { TextStyle } from '../react-tidbits'

type LinkProps = {
  value: ReactNode
  textStyle: TextStyle
  url?: string | { pathname: string; query: Record<string, string> }
}

export const ValueLink = ({
  value,
  textStyle,
  url,
}: LinkProps): JSX.Element => {
  return url ? (
    <Link href={url} textStyle={textStyle}>
      {value}
    </Link>
  ) : (
    <Text component='div' textStyle={textStyle}>
      {DEFAULT_FALLBACK_TEXT}
    </Text>
  )
}
