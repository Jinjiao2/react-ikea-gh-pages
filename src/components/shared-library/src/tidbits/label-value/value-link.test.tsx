/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import React from 'react'

import theme from '@tidbits/react-tidbits/theme'

import { ValueLink } from './value-link'

describe('ValueLink component', () => {
  test('renders correctly', async () => {
    const tree = render(
      <ThemeProvider theme={theme}>
        <ValueLink value='test val' textStyle='h1Regular' url='test.com' />
      </ThemeProvider>,
    )
    expect(tree).toMatchSnapshot()
  })
})
