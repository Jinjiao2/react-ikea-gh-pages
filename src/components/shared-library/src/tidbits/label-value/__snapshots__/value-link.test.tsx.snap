// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValueLink component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="Box-sc-18eybku-0 Link-sc-1tc8rtf-0 dsrvPM jwLdat"
        color="#0070C9"
        cursor="pointer"
        href="test.com"
      >
        test val
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="Box-sc-18eybku-0 Link-sc-1tc8rtf-0 dsrvPM jwLdat"
      color="#0070C9"
      cursor="pointer"
      href="test.com"
    >
      test val
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
